### PR TITLE
Permit to send the deauth request on the previous equipment.

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Send the deauth request to the previous switch the device was connected.
In the case you have 2 equipment, one AP and a dhcp that trigger a radius request and you want the deauth request to be sent only on the dhcp equipment. 

# Impacts
Deauth

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Deauth request can be made on the previous equipment the device was connected.

### CODE REVIEW

**Comment by julsemaan** [comment]
IMO this can get in for 11.1 but we need the form+admin to be rebased and adjusted

**Comment by extrafu** [comment]
@fdurand Please rebase and @satkunas review the UI code.

**Comment by fdurand** [comment]
rebase done

**Comment by satkunas** [comment]
UI code is OK to merge

File: html/pfappserver/lib/pfappserver/Form/Config/Switch.pm [hunk 1]
```diff
@@ -97,6 +97,15 @@ has_field 'useCoA' =>
              help => 'Use CoA when available to deauthenticate the user. When disabled, RADIUS Disconnect will be used instead if it is available.' },
   );
 
+has_field 'deauthOnPrevious' =>
+  (
+   type => 'Toggle',
+   label => 'Deauth On previous switch',
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Switch.pm]: Remove the label and help from here

File: html/pfappserver/root/src/views/Configuration/switchGroups/_components/TheForm.vue [hunk 2]
```diff
@@ -44,6 +44,11 @@
           :text="$i18n.t('Use CoA when available to deauthenticate the user. When disabled, RADIUS Disconnect will be used instead if it is available.')"
         />
 
+        <form-group-deauth-on-previous namespace="deauthOnPrevious"
+          :column-label="$i18n.t('Deauth on previous switch')"
+          :text="$i18n.t('This option parameter will allow you to do the deauthentication/CoA on the previous switch the device connected.')"
```
Comment (julsemaan) [inline html/pfappserver/root/src/views/Configuration/switchGroups/_components/TheForm.vue]:
```suggestion
          :text="$i18n.t('This option parameter will allow you to do the deauthentication/CoA on the previous switch where the device was connected.')"
```

File: html/pfappserver/root/src/views/Configuration/switches/_components/TheForm.vue [hunk 3]
```diff
@@ -47,6 +47,11 @@
           :text="$i18n.t('Use CoA when available to deauthenticate the user. When disabled, RADIUS Disconnect will be used instead if it is available.')"
         />
 
+        <form-group-deauth-on-previous namespace="deauthOnPrevious"
+          :column-label="$i18n.t('Deauth on previous switch')"
+          :text="$i18n.t('This option parameter will allow you to do the deauthentication/CoA on the previous switch the device connected.')"
```
Comment (julsemaan) [inline html/pfappserver/root/src/views/Configuration/switches/_components/TheForm.vue]:
```suggestion
          :text="$i18n.t('This option parameter will allow you to do the deauthentication/CoA on the previous switch whre the device was connected.')"
```

**Review by julsemaan — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
BUMP

**Pull request merged**